app/backend/src/couchers/servicers/api.py

```diff
@@ -574,7 +590,10 @@ def WriteReference(self, request, context):
             rating=request.rating,
         )
         with session_scope() as session:
-            if not session.query(User).filter(User.id == request.to_user_id).one_or_none():
+            if (
+                not session.query(User).filter(User.is_visible).filter(User.id == reference.from_user_id).one_or_none()
```

Comment:
No need to check `from_user`, they're guaranteed to exist and be visible (needs to be handled in auth)